oximeter/types/src/traits.rs

```diff
@@ -85,10 +87,12 @@ use std::ops::AddAssign;
 /// }
 /// ```
 ///
-/// **Important:** Deriving this trait is deprecated, and will be removed in the
-/// future. Instead, define your timeseries schema through the TOML format
-/// described in [the crate documentation](crate), and use the code
-/// generated by the `use_timeseries` macro.
+/// **Important:** Typically, you should define your timeseries schema through
+/// the TOML format described in [the crate documentation](crate), and use the
+/// code generated by the `use_timeseries` macro. However, if you need to handle
+/// custom sampling logic and must define custom field definitions not published
+/// as part of the target schema, you can implement this trait as long as you
+/// define the target as one of the field(s) in the struct.
```

Comment:
This is great, but we don't really need the `... as long as you...` bit. I'd remove that, but everything else looks good.